src/lib/services/addon-service.ts

```diff
@@ -259,6 +275,23 @@ export default class AddonService {
             );
         }
         const result = await this.addonStore.update(id, addonConfig);
+
+        if (
+            addonConfig.provider === 'webhook' &&
+            addonConfig.parameters.url &&
+            addonConfig.parameters.url.indexOf('://') > 0
+        ) {
+            const start = addonConfig.parameters.url.indexOf('://') + 3;
+            let end = addonConfig.parameters.url.indexOf('/', start);
+            if (end === -1) {
+                end = addonConfig.parameters.url.length;
+            }
+            const domain = addonConfig.parameters.url.substring(start, end);
```

Comment:
Duplicated code, can't this be extracted to a function?

Reply:
But also... why not just?
```
new URL(addonConfig.parameters.url).hostname
```

Reply:
Oh nice, did not even think about that. But obviously this exists 🙄